src/libtomahawk/resolvers/JSResolver.cpp

```diff
@@ -58,13 +59,15 @@ using namespace Tomahawk;
 
 JSResolver::JSResolver( const QString& accountId, const QString& scriptPath, const QStringList& additionalScriptPaths )
     : Tomahawk::ExternalResolverGui( scriptPath )
+    , ScriptPlugin( nullptr )
     , d_ptr( new JSResolverPrivate( this, accountId, scriptPath, additionalScriptPaths ) )
 {
     Q_D( JSResolver );
     tLog() << Q_FUNC_INFO << "Loading JS resolver:" << scriptPath;
 
-    d->engine = new ScriptEngine( this );
     d->name = QFileInfo( filePath() ).baseName();
+    d->scriptAccount.reset( new JSAccount( d->name ) );
```

Comment:
Not so easy right now, needs a few of the other functions and complicates things, I'll reconsider this when moving onward with the ScriptAccount refactor.